### ISSUE TITLE
ReinterpretArray and nested structure padding

### ISSUE THREAD

**Issue by chethega** [JuliaLang/julia#29605]
Unfortunately the current tests for structure padding [https://github.com/JuliaLang/julia/issues/25908](https://github.com/JuliaLang/julia/issues/25908) do not recurse down the type hierarchy, i.e. [https://github.com/JuliaLang/julia/pull/27877](https://github.com/JuliaLang/julia/pull/27877) is incomplete. Sorry, I should have thought about that possibility when @Keno first introduced the padding machinery :(

```
julia> struct padded
       a::Bool
       b::UInt32
       end

julia> struct wrapped
       x::padded
       end

julia> a=collect(1:2); r=reinterpret(wrapped, a); r[2]=r[1];

julia> a
2-element Array{Int64,1}:
        1
 16541441
```

**Comment by Keno** [comment]
yep, this was discussed and we should extend the padding mechanism to handle this, but we considered just getting the simple thing in first better than nothing.

**Comment by BioTurboNick** [comment]
The padding checks in #47116 properly recurses nested structures, but I don't believe the array reinterpret code is using it right now.

However, the example in the OP currently results in an error during `setindex!` on 1.10-rc2.

**Comment by BioTurboNick** [comment]
I think this is by design then - the example ReinterpretArray is deemed readable but not writable, because it's reading the padding properly. So this issue can probably be closed.

```
julia> dump(r)
Base.ReinterpretArray{wrapped, 1, Int64, Vector{Int64}, false}
  parent: Array{Int64}((2,)) [1, 2]
  readable: Bool true
  writable: Bool false
```

**Comment by adienes** [comment]
```
julia> r[2] = r[1]
ERROR: Padding of type wrapped is not compatible with type Int64.
```
